### ISSUE TITLE
Separate codebases

### ISSUE THREAD

**Issue by appgurueu** [Persistent-Kingdoms/persistent-factions#9]
This repo is a collection of different codebases, and as such, poorly manageable. Bugs stack up and we need to port every fix for MTG / factions etc here manually. So I propose that we: 

- Separate this codebase from MTG
- Separate it into many small mods and make the PK org have multiple subrepos
- Separate other mods from it, and don't apply changes directly in the code but using extension mods afterwards

This will mean quite some work at first, but will be extremely useful in the long term. We are running an outdated MTG with tons of outdated mods with many tweaks and fixes by us poured in, making it hard to merge. 

The benefits will be great: Straightforward licensing, less work, easy updates, merging, new features, and finally being able to concentrate on our neatly separated PK code.

**Comment by archfan7411** [comment]
Overall agreed, the prospect of separating custom(ized) mods into other PK repos is especially alluring. Unsure what you mean by "Separate this codebase from MTG" though, could you elaborate?

**Comment by appgurueu** [comment]
`kingdoms_game` extends `minetest_game`, so we should separate it

**Comment by archfan7411** [comment]
How would that be accomplished?

**Comment by appgurueu** [comment]
By having the MTG mods separated from this mods. You would always get the freshest ones by pulling them. [Texmex](https://forum.minetest.net/memberlist.php?mode=viewprofile&u=19662) is one example of a provider of [freshly chopped MTG mods](https://github.com/minetest-game-mods)

**Comment by appgurueu** [comment]
I'm not sure about the details, though. We need a way to manage a collection of mods as one game. I'd most likely go with submodules. But apparently there have been issues with that...

**Comment by appgurueu** [comment]
Other members: what do you think?
Pings in no specific order:
* @Cahrs
* @Coder12a
* @Elkien3
* @luk3yx

**Comment by luk3yx** [comment]
I suggest something similar to https://github.com/BlockySurvival/bls_mods/ (including the instructions on how to clone and update the repository).

**Comment by appgurueu** [comment]
Indeed, that's simply awesome. Conclusions: 
* Separate game and modpack, make a PK modpack
* Use submodules
* Proper Git instructions
* ~~Steal their repo~~

**Comment by appgurueu** [comment]
In general, BlockySurvival is a great example for how the entire organization could work. Amazing.

**Comment by ghost** [comment]
Hola, I am late to the party.

**Comment by ghost** [comment]
Hmmmmm.

**Comment by appgurueu** [comment]
First step done ! See our brand new [Minetest Game Modpack](https://github.com/Persistent-Kingdoms/minetest-game) :tada:

**Comment by appgurueu** [comment]
Second step done ! [Kingdoms Game](https://github.com/Persistent-Kingdoms/kingdoms-game) repo made (uses submodules as well)

**Comment by appgurueu** [comment]
Ideally, the final step, as I imagine it (probably by the end of the year), will be swapping out Persistent Factions by the Kingdoms Game, keeping the world. @archfan7411

**Comment by archfan7411** [comment]
Awesome! Just one minor thing: the games are `minetest_game` and `kingdoms_game` but you used dashes in place of underscores. Obviously very minor, it just bugs my OCD. Optional if you want to change it.

**Comment by appgurueu** [comment]
Yeah, the thing is, GitHub convention pretty much uses hyphens, as perceived by me. So I just adhere to it and use `a-b-c` instead of `a_b_c`. Different naming conventions. I would prefer the GH one as it's more visually balanced, the hyphen being in the middle of the line instead of at the bottom (it just looks neater, doesn't it?). ~~At least no lowerCamelCase.~~

**Comment by archfan7411** [comment]
At the risk of bikeshedding, `minetest_game` is "correct" because that's the actual name of the game :P
This is getting offtopic though, so I'll just drop it. Thanks again for going through all that work to make the repos!

**Comment by appgurueu** [comment]
Change made.
